pymeasure/instruments/keysight/keysightN7776C.py

```diff
@@ -0,0 +1,256 @@
+                                    values=WL_RANGE,
+                                    get_process=lambda v: v*1e9)
+
+    sweep_wl_start = Instrument.control('sour0:wav:swe:star?','sour0:wav:swe:star %fnm',
+                                    """ Start Wavelength (in nanometers) for a sweep.""",
+                                    validator=strict_range,
+                                    values=WL_RANGE,
+                                    get_process=lambda v: v*1e9)
+    sweep_wl_stop = Instrument.control('sour0:wav:swe:stop?','sour0:wav:swe:stop %fnm',
+                                    """ End Wavelength (in nanometers) for a sweep.""",
+                                    validator=strict_range,
+                                    values=WL_RANGE,
+                                    get_process=lambda v: v*1e9)
+
+    sweep_step = Instrument.control('sour0:wav:swe:step?','sour0:wav:swe:step %fnm',
+                                    """ Step width of th[e sweep (in nanometers).""",
+                                    validator=strict_range,
+                                    values=[0.0001,WL_RANGE[1]-WL_RANGE[0]])
+    sweep_speed = Instrument.control('sour0:wav:swe:speed?','sour0:wav:swe:speed %fnm/s',
+                                    """ Speed of the sweep (in nanometers per second).""",
+                                    validator=strict_discrete_set,
+                                    values=[0.5,1,50,80,200],
+                                    get_process=lambda v: v*1e9)
+    sweep_mode = Instrument.control('sour0:wav:swe:mode?','sour0:wav:swe:mode %s',
+                                    """ Sweep mode of the swept laser source """,
+                                    validator=strict_discrete_set,
+                                    values=['STEP','MAN','CONT'])
+    sweep_twoway = Instrument.control('sour0:wav:swe:rep?','sour0:wav:swe:rep %s',
+                                    """Sets the repeat mode. Applies in stepped,continuous and manual sweep mode.""",
+                                    validator=strict_discrete_set,
+                                    values=[True,False],
+                                    set_process=lambda v: ['ONEW','TWOW'][int(v)],
+                                    get_process=lambda v: bool(['ONEW','TWOW'].index(v)))
+    _sweep_check = Instrument.measurement('sour0:wav:swe:chec?',
+                                    """Returns whether the currently set sweep parameters (sweep mode, sweep start, stop, width, etc.) are consistent. If there is a
+                                    sweep configuration problem, the laser source is not able to pass a wavelength sweep.""")
+
+    sweep_points = Instrument.measurement('sour0:read:points? llog',
+                                    """Returns the number of datapoints that the :READout:DATA? command will return.""")
+    
+    sweep = Instrument.control('sour0:wav:swe?','sour0:wav:swe %g',
```

Comment:
Clearer: `sweep_state`? `sweep` could as well be a verb/command.

Reply:
Also, how is this used?

Reply:
Changed to sweep_state.

It is basically a state indicator giving either 0 (not running), 1 (running) or 2 (paused). Mostly it will be used for checking if a continuous sweep is finished, but other use cases are possible

Reply:
Thanks, best document those values in the docstring, and use a `strict_discrete_set` validator to make sure that only valid values get through. Trinary logic is tricky (can't easily be converted to true/false) - I'd suggest either staying with the integers, or map them to strings "stopped"/"running"/"paused". Consider how a user determines which is which.